allennlp/training/trainer.py

```diff
@@ -66,6 +68,7 @@ def __init__(
         local_rank: int = 0,
         world_size: int = 1,
         num_gradient_accumulation_steps: int = 1,
+        opt_level: Optional[str] = None,
```

Comment:
Can you add this to the docstring?

Reply:
Done, added a comment below.

Reply:
What about `amp_level`?

Reply:
Hmm, the also call this arg `opt_level` in `apex` (see [here](https://nvidia.github.io/apex/amp.html#opt-levels-and-properties)). Is there a good argument for calling it something different in AllenNLP?

Reply:
Yeah, cause amp is the main thing in apex. As far as I understand, `opt_level` means "option level". In AllenNLP there are many options. So, `amp_level` would mean "automatic mixed precision level"

Reply:
[Oh, it's actually "optimization level"](https://nvidia.github.io/apex/amp.html#opt-levels-and-properties). Still, I'd do `amp_opt_level`, cause there are many different options in `Trainer` and it can get confusing. What do you think?

Reply:
I would lean towards sticking with `opt_level` here, or perhaps `amp_opt_level`.  Switching the name of a parameter makes it harder for someone familiar with apex to know what's going on here.